### ISSUE TITLE
Upgrading to 5.0.0 breaks all RSpec tests

### ISSUE THREAD

**Issue by colinwarmstrong** [thoughtbot/factory_bot_rails#315]

**Comment by composerinteralia** [comment]
That may be related to https://github.com/thoughtbot/factory_bot_rails/issues/303.
Do you get a different error if you run just one test at a time?

**Comment by kofronpi** [comment]
Same here. I (finally) did the upgrade process from `factory_girl` to `factory_bot`, it upgraded factory_bot to 5.0.0 and I have the same issue. I'm trying downgrading it a bit

**Comment by composerinteralia** [comment]
@kofronpi if you are coming from factory_girl I would recommend upgrading to factory_bot 4.11.1 first so you can catch any deprecation warnings.

**Comment by kofronpi** [comment]
Yes, I downgraded to 4.10 (saw mention of the frozen array issue somewhere
else with 4.11) and it works now. Thanks !

**Comment by composerinteralia** [comment]
OK, if you do plan on upgrading later, my experience with that Frozen Array issue is that if you run just a single test you will get the "real" error message. I once traced through why we end up with a Frozen Array error instead of the underlying factory_bot error, if that helps at all: https://github.com/thoughtbot/factory_bot_rails/issues/303#issuecomment-434560625

**Comment by colinwarmstrong** [comment]
Ended up going back this morning to try and figure out the issue when upgrading.

So I ran an individual test like you suggested and got the "real error" message I was looking for:
```
NoMethodError:
  undefined method 'name' in 'award' factory
```

Apparently in the newest version of FactoryBot, factory attribute values must be defined using bracket notation like so:
```
FactoryBot.define do
  factory :award do
    name { 'Teamwork' }
  end
end
```
Without the brackets, you get an undefined method error for your attribute names.  When you get multiple errors of this kind happening, it triggers a Frozen Array error instead.

So using the curly-bracket notation in all of my factories allowed me to run all my tests again without issues.

**Comment by composerinteralia** [comment]
Yup, static attributes were deprecated in 4.11 and then removed in 5.0. The deprecation warning in 4.11 mentions a rubocop-rspec cop that can help change all the static attributes to dynamic ones. Let me know if you run into any other problems!
